### ISSUE TITLE
Augmented 0.9.5 is not working if firefox have set Cookies disabled

### ISSUE THREAD

**Issue by Killogy** [tfedor/AugmentedSteam#169]
I´m running Firefox 56.0.2 vanilla version, I have set Firefox Histoy setting at "Use Custom settings for history" and unchecked "Accept cookies from sites", then Augmented doesn´t work on Community pages  profile pages and store, store is working with 0.9.3.

I have cookies from sites I want to store cookies added to the explusion list so these sites can store cookies.

I have never any issues with previous extension Enhanced Steam, is working even with cookies disabled. Augmented 0.9.3 is working with cookies disabled so something in the latest 0.9.5 has changed.

**Comment by KarlCastle** [comment]
Somewhere between FF56 and FF65 the setting has moved to (emphasis mine):
> Privacy & Security > Content Blocking > Custom > All cookies *(will cause websites to break)*

On that setting, the extension does load (with errors) on the front page, though the front page itself has missing images because of cookie errors, and you can't log in.
<img src="https://user-images.githubusercontent.com/9707792/54125922-3e067e80-4456-11e9-8422-e3f18ed96b2d.jpg" alt="Welcome to Steam" width="100" />

Ditto Steam Community.
<img src="https://user-images.githubusercontent.com/9707792/54126102-b0775e80-4456-11e9-88a6-34f9779e3afe.png" alt="Welcome to Steam" width="100" />

The change (blocking website cookies no longer blocks `localStorage` in extensions) seems to be part of FF63? https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Privacy/Storage_access_policy

I'm not sure of the value of a storefront where you can't buy anything or a community site where you can't post?  At the moment we are still working on some features from Enhanced Steam 9.9 that haven't made it across yet.

**Comment by Killogy** [comment]
I tested using the portable version of Firefox 65, is working properly if Firefox is set to blocking cookies Privacy & Security > Content Blocking > Custom > All cookies (will cause websites to break), I added Steam domains to exception and I can log in.

But using firefox 56, Augmented stopped working after 0.9.3. 

Augmented 0.9.5 is working properly if cookies are enabled, but does not load if Firefox is set to cookies disabled, that´s the difference beetwen 0.9.3 and 0.9.5.

In summary:

Augmented 0.9.3 Does load even if cookies are disabled.

Augmented 0.9.5 Does not load if cookies are disabled, even if Steam domains has been added to the exclusion list, so these cookies are permitted to be stored.

**Comment by MxtOUT** [comment]
Any updates on this?

**Comment by Killogy** [comment]
@MxtOUT is almost working. when visiting store, profile or store game page Augmented load with errors **1 or 2 request(s) failed, check browser console for more information** but Augmented is loading Store game page features/links as ITAD, Metacritic.. and profile links, but ITAD info is not showing on Whislist section.

Store frontpage **1 request(s) failed, check browser console for more information**

`DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/background/background.js:11] early_access_appids  background.js:740:

DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/core.js:303] login  background.js:740:13

Object { error: true, message: "The operation is insecure.", action: "login", stack: "get@moz-extension://6d7b2794-5fe3-4…", localStack: "message/</<@moz-extension://6d7b279…" }  common.js:384:27

DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/background/background.js:11] currency  background.js:740:13

Couldn't load currency[object Object]  common.js:669:17

DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/background/background.js:11] rates  background.js:740:13

Failed to initialize Currency  common.js:697:17

Object { error: true, message: "The operation is insecure.", action: "rates", stack: "get@moz-extension://6d7b2794-5fe3-4…", localStack: "message/</<@moz-extension://6d7b279…" }  comm


Error: WebExtension context not found! ExtensionParent.jsm:778:13

Store page **2 request(s) failed, check browser console for more information**

`TypeError: result is undefined[Saber más]  store.js:1498:6

Object { error: true, message: "The operation is insecure.", action: "storepagedata", stack: "get@moz-extension://6d7b2794-5fe3-4…", localStack: "message/</<@moz-extension://6d7b279…" }  store.js:559:62
	AppPageClass/this.data< moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/content/store.js:559:62

DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/background/background.js:11] storepagedata  background.js:740:13

DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/core.js:303] login  background.js:740:13

Object { error: true, message: "The operation is insecure.", action: "login", stack: "get@moz-extension://6d7b2794-5fe3-4…", localStack: "message/</<@moz-extension://6d7b279…" }  common.js:384:27

DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/background/background.js:11] rates  background.js:740:13

Failed to initialize Currency  common.js:697:17

Object { error: true, message: "The operation is insecure.", action: "rates", stack: "get@moz-extension://6d7b2794-5fe3-4…", localStack: "message/</<@moz-extension://6d7b279…" }  common.js:698:17
`


Profile page: **1 request(s) failed, check browser console for more information**

`DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/background/background.js:11] early_access_appids  background.js:740:13

DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/background/background.js:11] profile  background.js:74

unreachable code after return statement[Saber más]  stickers.js:93:1

unreachable code after return statement[Saber más]  community.js:2137:8

DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/core.js:303] login  background.js:740:13

Object { error: true, message: "The operation is insecure.", action: "login", stack: "get@moz-extension://6d7b2794-5fe3-4…", localStack: "message/</<@moz-extension://6d7b279…" }  common.js:384:27

DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/background/background.js:11] currency  background.js:740:13

Couldn't load currency[object Object]  common.js:669:17

DOMException [SecurityError: "The operation is insecure."
code: 18
nsresult: 0x80530012
location: moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/background/background.js:11] rates  background.js:740:13

Failed to initialize Currency  common.js:697:17
	self.init/_promise< moz-extension://6d7b2794-5fe3-453e-9f16-a7d31932b638/js/content/common.js:697:17

Object { error: true, message: "The operation is insecure.", action: "rates", stack: "get@moz-extension://6d7b2794-5fe3-4…", localStack: "message/</<@moz-extension://6d7b279…" }  common.js:698:17`

**Comment by MxtOUT** [comment]
There has been a fix in 60a0a8b695d1837289ee10673ed9be353d8e45b2 regarding 3rd party cookies.
Regarding blocking all cookies:
I just tried using Steam with that setting and it's horrible.
If Steam isn't even working with that, Augmented Steam also shouldn't.

Thank you for your information though :)
